tools/babylon-status/pool_status.py

```diff
@@ -1,96 +1,196 @@
-        continue
-    #status_resources = pool.get('status', {}).get('resources', [])
-    label = 'poolboy.gpte.redhat.com/resource-pool-name='+pool['metadata']['name']
-    handles_resp = custom_objects_api.list_namespaced_custom_object(
-                   'poolboy.gpte.redhat.com',
-                   'v1',
-                   'poolboy',
-                   'resourcehandles',
-                   label_selector=label)
-    handles = handles_resp['items']
-    min_available = pool['spec']['minAvailable']
-    total = 0
-    available = 0
-    taken = 0
-
-    for handle in handles:
-        total = total + 1
-        ttotal = ttotal + 1
-
-        if 'resourceClaim' in handle['spec']:
-            taken = taken + 1
-            ttaken = ttaken + 1
-            continue
-
-        if 'resources' not in handle['spec']:
+core_v1_api = custom_objects_api = pool_pattern = None
+poolboy_domain = 'poolboy.gpte.redhat.com'
+poolboy_version = 'v1'
+poolboy_namespace = 'poolboy'
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="List and update Resource Pools")
+    parser.add_argument('--list',
+                        help='Lists all pools. [default: %(default)s]',
+                        action="store_true",
+                        default=True,
+                        required=False)
+
+    parser.add_argument('--pattern',
+                        dest="pool_pattern",
+                        help='Search pool by pattern',
+                        const='',
+                        nargs='?',
+                        required=False)
+
+    parser.add_argument('--pool-name',
+                        dest="pool_name",
+                        help='Pool Name',
+                        nargs=1,
+                        required=False)
+
+    parser.add_argument('--set-min',
+                        dest="set_min",
+                        help='Min replica',
+                        nargs=1,
+                        type=int,
+                        required=False)
+
+    args = parser.parse_args()
+
+    return args
+
+
+def search_pool():
+    # min_available = pool['spec']['minAvailable']
+    try:
+        pool_config = custom_objects_api.get_namespaced_custom_object(poolboy_domain, poolboy_version,
+                                                                      poolboy_namespace,
+                                                                      'resourcepools', pool_name)
+        return pool_config
+    except kubernetes.client.exceptions.ApiException as e:
+        if e.status == 404:
+            print(f"Resource Pool Not Found {pool_name}")
+            exit(-1)
+        else:
+            print(f"Error connect to K8s API {e}")
+            exit(-2)
+
+
+def set_min():
+    try:
+        resource_pool_data = search_pool()
+        if len(resource_pool_data) == 0:
+            return
+
+        previous_value = resource_pool_data['spec']['minAvailable']
+        if previous_value == pool_min:
```

Comment:
This use of a global variable seems not best practice. Shouldn't `pool_min` be passed as an argument to `set_min()`?